frontend/document/documentregistry.lua

```diff
@@ -26,6 +27,34 @@ function DocumentRegistry:addProvider(extension, mimetype, provider, weight)
     self.filetype_provider[extension] = true
 end
 
+function DocumentRegistry:getRandomFile(dir, opened, extension)
+    local DocSettings = require("docsettings")
+    if string.sub(dir, string.len(dir)) ~= "/" then
+        dir = dir .. "/"
+    end
+    local files = {}
+    local i = 0
+    math.randomseed(os.time())
```

Comment:
Wouldn't this be clearer just before where math.random is actually used? Afaik it's only if you want to give it a different seed that you'd want to wait a little to make it sufficiently distinct (in seconds for Lua's `os.time()`).

Reply:
Do you mean...
```
diff --git a/frontend/document/documentregistry.lua b/frontend/document/documentregistry.lua
index 37543151..b1421786 100644
--- a/frontend/document/documentregistry.lua
+++ b/frontend/document/documentregistry.lua
@@ -34,7 +34,6 @@ function DocumentRegistry:getRandomFile(dir, opened, extension)
     end
     local files = {}
     local i = 0
-    math.randomseed(os.time())
     local ok, iter, dir_obj = pcall(lfs.dir, dir)
     if ok then
         for entry in iter, dir_obj do
@@ -51,7 +50,7 @@ function DocumentRegistry:getRandomFile(dir, opened, extension)
     else
         return nil
     end
-    logger.info(files)
+    math.randomseed(os.time())
     return dir .. files[math.random(i)]
 end 
```

Reply:
Yes, indeed. But while my asking the question implies I think that, you can disagree of course. :-)